### PR TITLE
[ADF-1949 ] Allow customization of the dropdown menu on document picker

### PR DESCRIPTION
**Pull request by suzanadirla**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Please see Jira task https://issues.alfresco.com/jira/browse/ADF-1949

**What is the new behaviour?**
The dropdown from the document picker can display what list of sites the developer wants, as long as it is the right format.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/content-services/document-list/components/document-list.component.ts [hunk 1]
```diff
@@ -612,12 +619,12 @@ export class DocumentListComponent implements OnInit, OnChanges, AfterContentIni
...
                     }
                 };
-                this.onPageLoaded(page);
+                this.onPageLoaded(page, merge);
             })
             .catch(error => this.error.emit(error));
     }
 
-    private loadRecent(): void {
+    private loadRecent(merge: boolean = false): void {
```
Comment (eromano) [inline lib/content-services/document-list/components/document-list.component.ts]: please avoid use of flag arguments  for more informarion
https://martinfowler.com/bliki/FlagArgument.html

File: lib/content-services/site-dropdown/sites-dropdown.component.ts [hunk 2]
```diff
@@ -28,22 +28,28 @@ export class DropdownSitesComponent implements OnInit {
...
     public siteSelected: string;
 
     constructor(private sitesService: SitesApiService) {}
 
     ngOnInit() {
-        this.sitesService.getSites().subscribe((result) => {
-            this.siteList = result;
-        },
-        (error) => {});
+        // fetch sites data only if siteList input has not been given a value
```
Comment (eromano) [inline lib/content-services/site-dropdown/sites-dropdown.component.ts]: remove this comment

File: lib/content-services/document-list/components/document-list.component.ts [hunk 3]
```diff
@@ -839,6 +846,10 @@ export class DocumentListComponent implements OnInit, OnChanges, AfterContentIni
     }
 
     canNavigateFolder(node: MinimalNodeEntity): boolean {
```
Comment (eromano) [inline lib/content-services/document-list/components/document-list.component.ts]: can we refactor all this method to have only one return?

File: lib/content-services/document-list/components/document-list.component.ts (hunk 1 above)
Comment (DenisVuyka) [reply lib/content-services/document-list/components/document-list.component.ts]: @eromano the merge comes from the "infinite load" feature, we already have `reload(merge: boolean = false)` and `loadFolder(merge: boolean = false)`

File: lib/content-services/site-dropdown/sites-dropdown.component.ts (hunk 2 above)
Comment (suzanadirla) [reply lib/content-services/site-dropdown/sites-dropdown.component.ts]: I removed the comment and added a method with an explanatory name as you requested. Please review, thank you.

File: lib/content-services/document-list/components/document-list.component.ts (hunk 3 above)
Comment (suzanadirla) [reply lib/content-services/document-list/components/document-list.component.ts]: I removed my changes from this method and refactored the performNavigation method instead. Please review. Thank you, Eugenio.

File: lib/content-services/document-list/components/document-list.component.ts (hunk 1 above)
Comment (suzanadirla) [reply lib/content-services/document-list/components/document-list.component.ts]: As mentioned by Denys, I used the 'merge' to assure that the infinite load will work also on the custom source lists.

**Pull request merged**